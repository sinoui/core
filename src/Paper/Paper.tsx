import styled, { css } from 'styled-components';
import classNames from 'classnames';
import { transitions } from '@sinoui/theme';

export interface PaperProps {
  /**
   * 阴影高度
   */
  elevation?: number;
  /**
   * 是否显示圆角
   */
  square?: boolean;
  /**
   * 添加自定义类名
   */
  className?: string;
  /**
   * 轮廓模式
   */
  outlined?: boolean;
}

const normalCss = css<{ elevation?: number }>`
  box-shadow: ${({ theme, elevation = 1 }) => theme.shadows[elevation]};
`;

const outlinedCss = css`
  border: 1px solid ${({ theme }) => theme.palette.divider};
`;

const roundCss = css`
  border-radius: 4px;
`;

/**
 * 纸张
 */
const Paper = styled.div.attrs({
  className: classNames('sinoui-paper'),
})<PaperProps>`
  box-sizing: border-box;
  ${({ square }) => !square && roundCss};
  background-color: ${({ theme }) => theme.palette.background.paper};
  ${({ outlined }) => (outlined ? outlinedCss : normalCss)};
  transition: ${({
    theme: {
      transitions: { create, easing, duration },
    },
  }) =>
    create(['box-shadow'], {
      duration: duration.shorter,
      easing: easing.easeInOut,
    })};
`;

export default Paper;
