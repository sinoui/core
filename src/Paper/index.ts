import Paper, { PaperProps } from './Paper';

export { PaperProps };
export default Paper;
