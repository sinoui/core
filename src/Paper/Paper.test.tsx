import React from 'react';
import renderer from 'react-test-renderer';

import styled, { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import Paper from '@sinoui/core/Paper';

const PaperWrapper = styled(Paper)`
  width: 128px;
  height: 128px;
  margin: 8px;
`;

/**
 * Paper组件 测试
 */
describe('Paper组件 快照测试', () => {
  it('基本使用', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <PaperWrapper>Paper Content</PaperWrapper>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('设置阴影高度', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <PaperWrapper elevation={4}>Paper Content</PaperWrapper>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('设置是否显示方角', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <PaperWrapper square>Paper Content</PaperWrapper>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('设置轮廓模式', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={defaultTheme}>
          <PaperWrapper outlined>Paper Content</PaperWrapper>
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
