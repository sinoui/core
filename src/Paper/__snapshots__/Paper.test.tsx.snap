// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paper组件 快照测试 基本使用 1`] = `
<div
  className="Paper-t0j6j6-0 Papertest__PaperWrapper-sc-7j6t7q-0 hKMuMv sinoui-paper"
>
  Paper Content
</div>
`;

exports[`Paper组件 快照测试 设置是否显示圆角 1`] = `
<div
  className="Paper-t0j6j6-0 Papertest__PaperWrapper-sc-7j6t7q-0 kYuRKD sinoui-paper"
>
  Paper Content
</div>
`;

exports[`Paper组件 快照测试 设置轮廓模式 1`] = `
<div
  className="Paper-t0j6j6-0 Papertest__PaperWrapper-sc-7j6t7q-0 jAOYBL sinoui-paper"
>
  Paper Content
</div>
`;

exports[`Paper组件 快照测试 设置阴影高度 1`] = `
<div
  className="Paper-t0j6j6-0 Papertest__PaperWrapper-sc-7j6t7q-0 eHeunO sinoui-paper"
  elevation={4}
>
  Paper Content
</div>
`;
